packages/cli/src/oclif/commands/register.js

```diff
@@ -63,37 +95,82 @@ class RegisterCommand extends ZapierBaseCommand {
     }
   }
 
+  /**
+   * Prompts user for values that have not been provided
+   * Flags can heavily impact the behavior of this function
+   * @returns { appMeta: {object}, action: string }
+   */
   async _promptForAppMeta() {
     const appMeta = {};
 
+    const actionChoices = [
+      { name: 'Yes, update current integration', value: 'update' },
+      { name: 'No, register a new integration', value: 'register' },
+    ];
+
+    let action = actionChoices[1].value; // Default action is register
+
+    const linkedAppId = (await getLinkedAppConfig(undefined, false))?.id;
+    if (linkedAppId) {
+      console.info(colors.yellow(`${CURRENT_APP_FILE} file detected.`));
+      if (this.flags.yes) {
+        console.info(
+          colors.yellow(
+            `-y/--yes flag passed, updating current integration (ID: ${linkedAppId}).`
+          )
+        );
+        action = actionChoices[0].value;
+      } else {
+        action = await this.promptWithList(
+          `Would you like to update your current integration (ID: ${linkedAppId})?`,
+          actionChoices
+        );
+      }
+    }
+
+    if (action === 'update') {
+      this.startSpinner('Retrieving details for your integration');
+      this.app = await getWritableApp();
+      this.stopSpinner();
+    }
+
     appMeta.title = this.args.title?.trim();
     if (!appMeta.title) {
       appMeta.title = await this.prompt(
         'What is the title of your integration?',
-        { required: true }
+        {
+          required: true,
+          default: this.app?.title,
+        }
       );
     }
 
     appMeta.description = this.flags.desc?.trim();
     if (!appMeta.description) {
       appMeta.description = await this.prompt(
         `Please provide a sentence describing your app in ${MAX_DESCRIPTION_LENGTH} characters or less.`,
-        { required: true, charLimit: MAX_DESCRIPTION_LENGTH }
+        {
+          required: true,
+          charLimit: MAX_DESCRIPTION_LENGTH,
+          default: this.app?.description,
+        }
       );
     }
 
     appMeta.homepage_url = this.flags.url;
     if (!appMeta.homepage_url) {
       appMeta.homepage_url = await this.prompt(
-        'What is the homepage URL of your app? (optional)'
+        'What is the homepage URL of your app? (optional)',
+        { default: this.app?.homepage_url }
       );
     }
 
     appMeta.intention = this.flags.audience;
     if (!appMeta.intention) {
       appMeta.intention = await this.promptWithList(
         'Are you building a public or private integration?',
-        this.config.enumFieldChoices.intention
+        this.config.enumFieldChoices.intention,
+        { default: this.app?.intention }
```

Comment:
@gregilo Why don't we have `required: true` for intention? Perhaps, I'm missing something?

Reply:
Learned that we don’t need it for the `promptWithList` fields because users have to make a selection out of the choices presented and no way out.